connect/api/src/main/java/org/apache/kafka/connect/sink/SinkTask.java

```diff
@@ -88,15 +88,15 @@ public void initialize(SinkTaskContext context) {
     public abstract void start(Map<String, String> props);
 
     /**
-     * Put the records in the sink. Usually this should send the records to the sink asynchronously
-     * and immediately return.
-     *
+     * Put the records in the sink. This should either write them to the downstream system or batch them for
+     * later writing
```

Comment:
I agree that it's not necessary for this javadoc to prescribe asynchronous behavior, but it should certainly point out the pitfall that asynchronous users need to take special care.
```suggestion
     * Put the records in the sink. If this method returns before the records are durably written,
     * the task must implement {@link #flush(Map)} or {@link #preCommit(Map)} to ensure that
     * only durably written record offsets are committed, and that no records are dropped during failures.
```